packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -604,20 +605,33 @@ export function deferredUpdates<A>(fn: () => A): A {
   return runWithPriority(NormalPriority, fn);
 }
 
+let interactiveEventTimeStamp = 0;
+
 export function interactiveUpdates<A, B, C, R>(
   fn: (A, B, C) => R,
   a: A,
   b: B,
   c: C,
 ): R {
-  if (workPhase === NotWorking) {
-    // TODO: Remove this call. Instead of doing this automatically, the caller
-    // should explicitly call flushInteractiveUpdates.
-    flushPendingDiscreteUpdates();
+  const currentEvent = typeof window !== 'undefined' && window.event;
```

Comment:
Let's lift this logic into React DOM/Fire. The renderer can export two methods: `interactiveUpdates` and `flushInteractiveUpdates`.

Also maybe we should go with "discrete"? We've been using both but I kinda like "discrete" more. Not required to decide on that in this PR, though.

Reply:
Done :)